src/webui/api/appcontroller.cpp

```diff
@@ -1083,6 +1085,24 @@ void AppController::defaultSavePathAction()
     setResult(BitTorrent::Session::instance()->savePath().toString());
 }
 
+void AppController::getDirectoriesAction()
```

Comment:
@pktiuk 
Since you haven't replied to https://github.com/qbittorrent/qBittorrent/pull/20314#pullrequestreview-1860844769 I'll ask here.

In the future, wouldn't there be expectations of listing files under some directory? Would you consider making this API endpoint more generic by having a 'mode' parameter? ('file mode' or 'dir mode').
Or there should be another API endpoint just for listing files? but isn't these 2 endpoints similar and it would be cumbersome for the webAPI user to differentiate them? (compared to using a 'mode' parameter)

Reply:
I think that it is a good idea.   
Adding a mode would make using much easier in cases when we would like to get directories and files. (there should be 3 modes: `files`, `directories` and `all`).